tests/system-tests/internal/supporttools/tcpdump.go

```diff
@@ -0,0 +1,543 @@
+		apiClient,
+		deployContainerCfg,
+		stDeploymentName,
+		stNamespace,
+		supportToolsDeploySAName,
+		scheduleOnHosts,
+		deployLabelsMap)
+
+	glog.V(100).Infof("Creating deployment")
+
+	stDeployment, err = stDeployment.CreateAndWaitUntilReady(5 * time.Minute)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create deployment %s in namespace %s: %w",
+			stDeploymentName, stNamespace, err)
+	}
+
+	if stDeployment == nil {
+		return nil, fmt.Errorf("failed to create deployment %s", stDeploymentName)
+	}
+
+	return stDeployment, err
+}
+
+func defineTCPDumpContainer(
+	cImage,
+	packetCaptureInterface,
+	packetCaptureProtocol string) *pod.ContainerBuilder {
+	cName := "tcpdump"
+
+	cmd := tcpCaptureScript
+	if packetCaptureProtocol == udpProtocol {
+		cmd = udpCaptureScript
+	}
+
+	cCmd := []string{
+		"/bin/bash",
+		"-c",
+		fmt.Sprintf(cmd, packetCaptureInterface),
+	}
+
+	glog.V(100).Infof("Creating container %q", cName)
+
+	deployContainer := pod.NewContainerBuilder(cName, cImage, cCmd)
+
+	glog.V(100).Infof("Defining SecurityContext")
+
+	var trueFlag = true
+
+	userUID := new(int64)
+
+	*userUID = 0
+
+	securityContext := &corev1.SecurityContext{
+		RunAsUser:  userUID,
+		Privileged: &trueFlag,
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{
+				"SETFCAP",
+				"CAP_NET_RAW",
+				"CAP_NET_ADMIN",
+			},
+		},
+	}
+
+	glog.V(100).Infof("Setting SecurityContext")
+
+	deployContainer = deployContainer.WithSecurityContext(securityContext)
+
+	glog.V(100).Infof("Dropping ALL security capability")
+
+	deployContainer = deployContainer.WithDropSecurityCapabilities([]string{"ALL"}, true)
+
+	glog.V(100).Infof("Enable TTY and Stdin; needed for immediate log propagation")
+
+	deployContainer = deployContainer.WithTTY(true).WithStdin(true)
+
+	glog.V(100).Infof("%q container's  definition:\n%#v", cName, deployContainer)
+
+	return deployContainer
+}
+
+// SendTrafficCheckTCPDumpLogs is a wrapper function to reduce code duplication when probing for target IPs.
+func SendTrafficCheckTCPDumpLogs(
+	apiClient *clients.Settings,
+	proberPodObj *pod.Builder,
+	tcpdumpPodNamespace,
+	tcpdumpPodSelectorLabel,
+	targetHost,
+	targetProtocol,
+	targetPort string) error {
+	timeout := 3 * time.Minute
+	err := wait.PollUntilContextTimeout(
+		context.TODO(),
+		3*time.Second,
+		timeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			glog.V(100).Infof("Verifying that the expected outbound request can be seen in the tcpdump logs")
+
+			result, err := sendProbeAndCheckTCPDumpLogs(
+				apiClient,
+				proberPodObj,
+				tcpdumpPodNamespace,
+				tcpdumpPodSelectorLabel,
+				targetProtocol,
+				targetHost,
+				targetPort)
+
+			if err == nil && result {
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+	if err != nil {
+		return fmt.Errorf("expected request was not found in the packet sniffer logs; %w", err)
```

Comment:
it worth logging an error

Reply:
ufff, I have many like this :( give me few mins